bdk_chain/src/keychain/keychain_txout_index.rs

```diff
@@ -253,6 +253,38 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
             .map(|((_, derivation_index), spk)| (*derivation_index, spk))
     }
 
+    /// Convenience method to get [`revealed_and_lookahead_spks_of_keychain`]
+    /// of all keychains.
+    ///
+    /// [`revealed_and_lookahead_spks_of_keychain`]: Self::revealed_and_lookahead_spks_of_keychain
+    pub fn revealed_and_lookahead_spks_of_all_keychains(
+        &self,
+    ) -> BTreeMap<K, impl Iterator<Item = (u32, &Script)> + Clone> {
+        self.keychains
+            .keys()
+            .map(|keychain| {
+                (
+                    keychain.clone(),
+                    self.revealed_and_lookahead_spks_of_keychain(keychain),
+                )
+            })
+            .collect()
+    }
+
+    /// Iterates over the script pubkeys revealed and the ones contained in
+    /// the lookahead under `keychain`.
+    pub fn revealed_and_lookahead_spks_of_keychain(
+        &self,
+        keychain: &K,
+    ) -> impl DoubleEndedIterator<Item = (u32, &Script)> + Clone {
+        let lookahead = self.lookahead.get(keychain).unwrap_or(&0);
+        let next_index = self.last_revealed.get(keychain).map_or(0, |v| *v + 1) + lookahead;
+        self.inner
+            .all_spks()
+            .range((keychain.clone(), u32::MIN)..(keychain.clone(), next_index))
+            .map(|((_, derivation_index), spk)| (*derivation_index, spk))
+    }
```

Comment:
I might be wrong, but maybe it is useful to error in case of non existing keychain than give back an empty? 

We don't enforce this behavior consistently yet. In `spks_of_keychain` it is an `expect` so the call panics for wrong keychain. While in `revealed_spks_of_keychain` it doesn't and gives empty iterator. 

I am not sure whats the right answer here, but to me it seems better to just panic early in this particluar case because it's clearly an user error?